cirq/work/sampler.py

```diff
@@ -133,6 +138,126 @@ def sample(
 
         return pd.concat(results)
 
+    def sample_expectation_values(
+        self,
+        program: 'cirq.Circuit',
+        observables: Union['cirq.PauliSumLike', List['cirq.PauliSumLike']],
+        *,
+        num_samples: int,
+        params: 'cirq.Sweepable' = None,
+        permit_terminal_measurements: bool = False,
+    ) -> List[List[float]]:
+        """Calculates estimated expectation values from samples of a circuit.
+
+        This method can be run on any device or simulator that supports circuit
+        sampling. Compare with `simulate_expectation_values` in simulator.py,
+        which is limited to simulators but provides exact results.
+
+        Args:
+            program: The circuit to simulate.
+            observables: A list of observables for which to calculate
+                expectation values.
+            num_samples: The number of samples to take. Increasing this value
+                increases the accuracy of the estimate.
+            params: Parameters to run with the program.
+            permit_terminal_measurements: If the provided circuit ends in a
+                measurement, this method will generate an error unless this
+                is set to True. This is meant to prevent measurements from
+                ruining expectation value calculations.
+
+        Returns:
+            A list of expectation-value lists. The outer index determines the
+            sweep, and the inner index determines the observable. For instance,
+            results[1][3] would select the fourth observable measured in the
+            second sweep.
+        """
+        if num_samples <= 0:
+            raise ValueError(
+                f'Expectation values require at least one sample. Received: {num_samples}.'
+            )
+        if not observables:
+            raise ValueError('At least one observable must be provided.')
+        if not permit_terminal_measurements and program.are_any_measurements_terminal():
+            raise ValueError(
+                'Provided circuit has terminal measurements, which may '
+                'skew expectation values. If this is intentional, set '
+                'permit_terminal_measurements=True.'
+            )
+        qubits = ops.QubitOrder.DEFAULT.order_for(program.all_qubits())
+        qmap = {q: i for i, q in enumerate(qubits)}
+        num_qubits = len(qubits)
+        psums = (
+            [ops.PauliSum.wrap(o) for o in observables]
+            if isinstance(observables, List)
+            else [ops.PauliSum.wrap(observables)]
+        )
+
+        obs_pstrings: List[ops.PauliString] = []
+        for psum in psums:
+            pstring_aggregate = {}
+            for pstring in psum:
+                pstring_aggregate.update(pstring.items())
+            obs_pstrings.append(ops.PauliString(pstring_aggregate))
```

Comment:
what is this section doing? Factor out into a private helper function?

Reply:
The purpose of this is to get `InitObsSettings` for each `PauliSum`. Since `observables_to_settings` only accepts `PauliString`s, I perform this conversion for each `PauliSum` provided.

If it makes sense to you, I could fold this into `observables_to_settings` and have it accept a mix of `PauliString` and `PauliSum` as observables.